src/fhost_fpga.py

```diff
@@ -615,14 +656,30 @@ def get_pfb_snapshot(self, pol=0):
             p_data.append(complex(r0_to_r3['r3'][ctr], i3['i3'][ctr]))
         return p_data
 
-    def qdr_okay(self):
+    def qdr_okay(self, wait_time=1):
         """
-        Checks if parity bits on f-eng are zero
-        :return: True/False
+        Checks if parity bits on f-eng are consistent.
+        :param: wait_time : Float or None : If not None, fetch qdr parity bit,
+            wait this long, and fetch a second value; Else, use last read value
+            from cache. Value is not cached if wait_time is None.
+        :return: True if QDR is okay
         """
-        for cnt in range(0, 1):
-            err = self.registers.ct_ctrs.read()['data']['ct_parerr_cnt%d' % cnt]
-            if err == 0:
+        # ToDo: Should this be hard-coded?
```

Comment:
No, we should probably use int(configd['fengine']['f_per_fpga']) - or store that in the initialise stage and then use it here.